src/main/java/io/lumigo/core/utils/AwsUtils.java

```diff
@@ -47,6 +50,24 @@ public static TriggeredBy extractTriggeredByFromEvent(Object event) {
                         && ((DynamodbEvent) event).getRecords().size() > 0) {
                     triggeredBy.setArn(
                             ((DynamodbEvent) event).getRecords().get(0).getEventSourceARN());
+                    DynamodbEvent.DynamodbStreamRecord firstRecord =
+                            ((DynamodbEvent) event).getRecords().get(0);
+                    triggeredBy.setArn(firstRecord.getEventSourceARN());
+                    if (firstRecord.getDynamodb() != null
+                            && firstRecord.getDynamodb().getApproximateCreationDateTime() != null) {
+                        triggeredBy.setApproxEventCreationTime(
+                                firstRecord
+                                        .getDynamodb()
+                                        .getApproximateCreationDateTime()
+                                        .getTime());
+                    }
+                    List<String> messageIds =
+                            ((DynamodbEvent) event)
+                                    .getRecords().stream()
+                                            .map(AwsUtils::extractMessageIdFromDynamodbRecord)
```

Comment:
How many values can return here? maybe we need a hard limit

Reply:
It's configurable by the user, and the default is 50.
But you're right, I'll limit it with 50 just to not get exploded :)